lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -153,36 +155,35 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar by the specified ID.
  *
- * @param int $drupal_id Drupal user id.
- * @return object
+ * @param  string $id
+ * @param  string $type
+ * @return DoSomething\Northstar\Resources\NorthstarUser|null
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_user($id, $type = 'drupal_id') {
+  $cache_bin = 'cache_dosomething_northstar';
+  $cache_key = 'northstar_user_'.$type.'='.$id;
+
+  $northstar_user = cache_get($cache_key, $cache_bin);
 
-  if ($northstar_user_info) {
-    $northstar_user_info = $northstar_user_info->data;
+  if ($northstar_user) {
+    return $northstar_user->data;
   }
-  else {
-    $client = _dosomething_northstar_build_http_client();
 
-    $northstar_user_info = drupal_http_request($client['base_url'] . '/users/drupal_id/' . $drupal_id, array(
-      'headers' => $client['headers'],
-      'method' => 'GET',
-      ));
+  $northstar_user = dosomething_northstar_client()->getUser($type, $id);
 
-      $northstar_user_info = $northstar_user_info->data;
+  if ($northstar_user) {
+    cache_set($cache_key, $northstar_user, $cache_bin, REQUEST_TIME + 60*60*24);
```

Comment:
Since this will be a `NorthstarUser` rather than an array, we may need to add a `->toArray()` method to the response class to serialize before caching here.